src/main/java/org/folio/services/budget/BudgetService.java

```diff
@@ -129,13 +129,20 @@ private List<Error> checkRemainingEncumbrance(SharedBudget budget) {
     BigDecimal expenditures = BigDecimal.valueOf(budget.getExpenditures());
     BigDecimal credits = BigDecimal.valueOf(budget.getCredits());
     BigDecimal awaitingPayment = BigDecimal.valueOf(budget.getAwaitingPayment());
-    BigDecimal totalFunding = BigDecimal.valueOf(budget.getTotalFunding());
+    BigDecimal allocated = BigDecimal.valueOf(budget.getAllocated());
+    BigDecimal netTransfers = BigDecimal.valueOf(budget.getNetTransfers());
 
-    //[remaining amount we can encumber] = (allocated * allowableEncumbered) - (encumbered + awaitingPayment + expended)
+    // [remaining amount we can encumber] = (allocated + netTransfers) * allowableEncumbered - (encumbered + awaitingPayment + expended - credits)
+    // calculations should be consistent with BatchTransactionChecks in mod-finance-storage
     if (budget.getAllowableEncumbrance() != null) {
       log.info("checkRemainingEncumbrance:: Budget '{}' allowable encumbrance is not null", budget.getId());
-      BigDecimal newAllowableEncumbrance = BigDecimal.valueOf(budget.getAllowableEncumbrance()).movePointLeft(2);
-      if (totalFunding.multiply(newAllowableEncumbrance).compareTo(encumbered.add(awaitingPayment).add(expenditures).subtract(credits)) < 0) {
+      BigDecimal allowableEncumbrance = BigDecimal.valueOf(budget.getAllowableEncumbrance()).movePointLeft(2);
+      BigDecimal totalFunding = allocated.add(netTransfers);
+      // unavailable amount shouldn't be negative
```

Comment:
We can remove such comments when method names are self explanatory. E.g `zeroIfNegative`.

Reply:
agree, but remained with the same reason to keep the same code as in mod-finance-storage